From d6733e9f411bff7380bb10a1ac2709008b9bd5c1 Mon Sep 17 00:00:00 2001
From: Simon Horman <horms+renesas@verge.net.au>
Date: Wed, 3 Jul 2013 13:38:26 +0900
Subject: [PATCH 0959/1083] Revert "drm/rcar-du: Fix plane index wrap-around
 for multi-planar overlays"

This reverts commit d6336c3042398bff3bb641d32ae08e46672322aa.
(cherry picked from commit 39b024b06976a5915852bf2006a098399c1e0e2d)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index ca80939..22b87ec 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -142,16 +142,13 @@ void rcar_du_crtc_update_planes(struct drm_crtc *crtc)
 
 	for (i = 0; i < num_planes; ++i) {
 		struct rcar_du_plane *plane = planes[i];
-		unsigned int index = plane->hwindex;
 
 		prio -= 4;
-		dspr |= (index + 1) << prio;
+		dspr |= (plane->hwindex + 1) << prio;
 
 		if (plane->format->planes == 2) {
-			index = (index + 1) % 8;
-
 			prio -= 4;
-			dspr |= (index + 1) << prio;
+			dspr |= (plane->hwindex + 2) << prio;
 		}
 	}
 
-- 
1.8.3.2

